### PR TITLE
pkp/pkp-lib#6251 Add isPreprintOf relation to DOI exports

### PR DESCRIPTION
**Pull request by ajnyga**
https://github.com/pkp/pkp-lib/issues/6251

### CODE REVIEW

**Comment by ajnyga** [comment]
@ewhanson @NateWr 

This is an additional change to the OPS Crossref plugin to introduce isPreprintOf relation (link to the version of record DOI).

There is an older pr for this, but it was delayed because of the ongoing DOI work. See the discussion here where Nate suggest how the OPS DOI plugin should force versioning of DOIs if Crossref is used for registering: https://github.com/pkp/ops/pull/103/files. However, now I've just heard from Alex that this could change. But that is probably something that still needs to be resolved.

In any case this should be merged first, I can rebase this pr after that if needed: https://github.com/pkp/crossref-ops/pull/9

File: filter/PreprintCrossrefXmlFilter.inc.php [hunk 1]
```diff
@@ -302,21 +304,32 @@ public function createDOIDataNode($doc, $doi, $url)
      *
      * @param $doc DOMDocument
      * @param $parentDoi string
+     * @param $vorDoi string
      *
      * @return DOMElement
      */
-    public function createRelationsDataNode($doc, $parentDoi)
+    public function createRelationsDataNode($doc, $parentDoi, $vorDoi)
```
Comment (NateWr) [inline filter/PreprintCrossrefXmlFilter.inc.php:311]: Can you split this into two methods: `createParentDoiNode()` and `createVorDoiNode()`?

**Review by NateWr — CHANGES_REQUESTED** [review]
Looks good. Just the one change to code organisation.

File: filter/PreprintCrossrefXmlFilter.inc.php (hunk 1 above)
Comment (ajnyga) [reply filter/PreprintCrossrefXmlFilter.inc.php:311]: Are you thinking that those would be called inside `createRelationsDataNode` or inside `createPostedContentNode`?

Comment (NateWr) [reply filter/PreprintCrossrefXmlFilter.inc.php:311]: Inside `createPostedContentNode`

**Comment by ajnyga** [comment]
@NateWr split the function to two parts as requested.

Since https://github.com/pkp/crossref-ops/pull/9 is touching the same code, I will wait for this to be merged first.

**Review by NateWr — APPROVED** [review]

**Comment by NateWr** [comment]
Looks good. Are there any other pull requests that need to be merged simultaneously with this one?

**Comment by ajnyga** [comment]
hmm, why did I not see an alert about this...

There is another one, but it would be easier if this would be merged first and I could rebase the other and pr it then.

**Pull request merged**